### PR TITLE
InfoBoxes/Content: add Alternate 2 GR

### LINKED ISSUE #616: Some Alternate infoboxes are missing
Hello,

Currently using XCSoar 7.6

The following infoboxes are missing for alternates :
- Alternate 2 GR (glide ratio)
- Altitude difference to alternate 1
- Altitude difference to alternate 2

Kind regards

### PR DESCRIPTION
**Pull request by scls19fr**
Brief summary of the changes
----------------------------

Add Alternate 2 GR infobox

Related issues and discussions
------------------------------

Closes #616

### CODE REVIEW

File: src/InfoBoxes/Content/Type.hpp [hunk 1]
```diff
@@ -104,6 +104,7 @@ namespace InfoBoxFactory
     e_Alternate_1_Name, /* Displays name and bearing to the best alternate landing location */
     e_Alternate_2_Name, /* Displays name and bearing to the second alternate landing location */
     e_Alternate_1_GR, /* Geometric gradient to the arrival height above the best alternate. This is not adjusted for total energy */
+    e_Alternate_2_GR, /* Geometric gradient to the arrival height above the second alternate. This is not adjusted for total energy */
```
Comment (MaxKellermann) [inline src/InfoBoxes/Content/Type.hpp:107]: Inserting this in the middle breaks existing configuration files

Comment (scls19fr) [reply src/InfoBoxes/Content/Type.hpp]: Sorry about that I didn't noticed this side effect. This line is now moved before `e_NUM_TYPES`

File: src/InfoBoxes/Content/Factory.cpp [hunk 2]
```diff
@@ -702,6 +702,14 @@ static constexpr MetaData meta_data[] = {
     IBFHelperInt<InfoBoxContentAlternateGR, 0>::Create,
   },
 
+  // e_Alternate_2_GR
```
Comment (MaxKellermann) [inline src/InfoBoxes/Content/Factory.cpp:705]: This is the wrong array position.

**Comment by MaxKellermann** [comment]
If you do fixups in an unmerged PR, please amend the existing commits.

**Comment by scls19fr** [comment]
By amending, do you mean squashing commits ?

**Comment by MaxKellermann** [comment]
Amend and squash are different things. Squash combines two commits into one. Amend re-edits an existing commit (`git commit --amend`). Though I never use `--amend`, I use stgit instead which has its own fantastic workflow (https://stacked-git.github.io/)

**Comment by scls19fr** [comment]
Not sure I can do more currently... because I was compiling with Linux Debian 10 in a VMware workstation 16 Player VM. Unfortunately my VM is full (and I can't see a way to change virtual hard disk size... except creating a new one with a full development install). 
10 GB is too short with LXQt, Visual Studio Code and all libraries required by XCSoar

**Comment by MaxKellermann** [comment]
Now you posted a merge commit, but a merge commit doesn't belong in a PR.

**Comment by scls19fr** [comment]
There was a merge conflict (because of NEWS.TXT) which I fixed directly from Github interface.

**Comment by MaxKellermann** [comment]
Don't merge. Rebase, but never merge. Merging is what I do.

**Comment by lordfolken** [comment]
> Not sure I can do more currently... because I was compiling with Linux Debian 10 in a VMware workstation 16 Player VM. Unfortunately my VM is full (and I can't see a way to change virtual hard disk size... except creating a new one with a full development install).
We have a vagrant box that you can start in ./ide/vagrant/Vagrantfile this works with virtualbox on windows. Alternatively we have a docker container that can compile xcsoar under ./ide/docker/README.md

**Comment by lordfolken** [comment]
can you please redo the commit? 
* Pull a fresh copy of master or git pull --rebase
* Do your changes in separate commits
   * doc
   * infobox
* push again with force

**Comment by MaxKellermann** [comment]
Actually, if one commit changes/adds a feature, the documentation belongs in the same commit. Doing a separate commit is ok, but naturally, this is just one change.

**Comment by MaxKellermann** [comment]
I did the fixup myself, here's the commit:
02fb01d884a71cbd5b97401fbd95466dc1dcd508

**Pull request closed without merging**